runtime/compiler/codegen/J9CodeGenerator.cpp

```diff
@@ -2166,6 +2170,25 @@ J9::CodeGenerator::doInstructionSelection()
       if (doEvaluation)
          self()->evaluate(node);
 
+      if (opCode == TR::BBEnd)
```

Comment:
Does this code need to be guarded with `self()->comp()->getOption(TR_SplitWarmAndColdBlocks)` too?   It seems only useful for that optimization.